### PR TITLE
Use site ID to prevent self-connection.

### PR DESCRIPTION
**Pull request by mgoulish**
No change to the edge connectivity test -- let it run as unit test for now, per Nico.

### CODE REVIEW

**Review by nicob87 — APPROVED** [review]
Nice!

File: client/edge_connectivity_test.go [hunk 1]
```diff
@@ -0,0 +1,413 @@
+package client
```
Comment (ajssmith) [inline client/edge_connectivity_test.go:1]: Per the discussion, i thought this was moving over to test/integration

**Review by ajssmith — CHANGES_REQUESTED** [review]
This looks good. Wasn't the edge test moving locale?

Can you please also edit the pr title to its original?

File: client/edge_connectivity_test.go (hunk 1 above)
Comment (mgoulish) [reply client/edge_connectivity_test.go:1]: I talked to Nicolas about that and he ended up deciding that it should stay where it is.
I'm afraid I don't recall the reasoning now...

**Review by ajssmith — APPROVED** [review]
lgtm

Next step will be to refactor for ClusterTestRunner

**Comment by mgoulish** [comment]
closing in favor of 249 -- rebased.

**Pull request closed without merging**